components/outputs/plot-matplotlib.qmd

```diff
@@ -0,0 +1,93 @@
+                  ax.set_ylabel("Density")
+
+                  return fig  #<<
+
+
+          app = App(app_ui, server, debug=True)
+
+      relevantfunctions:
+        - title: ui.output_plot
+          href: https://shiny.posit.co/py/api/ui.output_plot.html
+          signature: ui.output_plot(id, width='100%', height='400px', *, inline=False, click=False, dblclick=False, hover=False, brush=False, fill=MISSING)
+        - title: "@render.plot"
+          href: https://shiny.posit.co/py/api/render.plot.html
+          signature: render.plot(_fn=None, *, alt=None, width=MISSING, height=MISSING, **kwargs)
+        - title: matplotlib.pyplot.subplots
+          href: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html
+          signature: matplotlib.pyplot.subplots(nrows=1, ncols=1, *, sharex=False, sharey=False, squeeze=True, width_ratios=None, height_ratios=None, subplot_kw=None, gridspec_kw=None, **fig_kw)
+        - title: matplotlib.axes.Axes.hist
+          href: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.hist.html
+          signature: Axes.hist(x, bins=None, range=None, density=False, weights=None, cumulative=False, bottom=None, histtype='bar', align='mid', orientation='vertical', rwidth=None, log=False, color=None, label=None, stacked=False, *, data=None, **kwargs)
+
+      details: |
+
+              <p>The Matplotlib is a library that can be used to create plots.</p>
+
+              <p>Follow three steps to display a Matplotlib figure in your app:</p>
+
+              <ol>
+                <li>
+                Add `ui.output_plot()` to the UI of your app to create a div in which to display the figure.
+                Where you call this function will determine where the figure will appear within the layout of the app.
+                </li>
+                <li>
+                Define a function within the `server()` function that creates the figure and
+                is wrapped around with the `@render.plot()` decorator to return the figure.
+                If your function calls reactive values, Shiny will update your figure whenever those values change,
+                in a [reactive fashion](https://shiny.posit.co/py/docs/reactive-programming.html).
+                </li>
+                  <ul>
+                    <li>The name of the decorated function (or `@output(id=...)`) should match the `id` of a `output_plot()` container.</li>
+                  </ul>
+
+                To notify Shiny that your function returns a reactive figure to display, decorate the function with:
+
+                <ul>
+                  <li>`@render.plot()`</li>
+                </ul>
+
+                <li>Pass `ui.output_plot()` the name of the function that builds the figure for the `input` argument,
+                as a character string,
+                to indicate which figure should be displayed in the div.
+                </li>
+              </ol>
+
```

Comment:
It might be useful to include a See also section, and then link to the other plot pages. I've been doing it like [this](https://github.com/posit-dev/py-shiny-site/pull/15/files#diff-c04103c288e8de2e806e3bff96a5971f36552997f5d52670b678456ec4a12767R84).(Greg said he can add links in later).